core/services/functions/listener.go

```diff
@@ -280,6 +283,42 @@ func (l *FunctionsListener) getMaxSecretsSize(flags RequestFlags) uint32 {
 	return l.pluginConfig.MaxSecretsSizesList[idx]
 }
 
+func (l *FunctionsListener) HandleOffchainRequest(ctx context.Context, request *OffchainRequest) error {
+	if request == nil {
+		return errors.New("HandleOffchainRequest: received nil request")
+	}
+	if len(request.RequestId) != RequestIDLength {
+		return fmt.Errorf("HandleOffchainRequest: invalid request ID length %d", len(request.RequestId))
+	}
+
```

Comment:
Should will do other validations here? Or do we expect that this will be handled by the gateway?
Such as enforce that subscriptionOwner & senderAddress are filled

Reply:
Some of these are done by the ORM but let me check if we should add anything here.

Reply:
Done